todo_app/app.py

```diff
@@ -1,29 +1,109 @@
-from flask import Flask, redirect, render_template, request
+import os
+import requests
+from flask import Flask, abort, redirect, render_template, request
+from flask_login import LoginManager, current_user, login_required, login_user
 from todo_app.data.cosmosdb_items import add_item, get_items, complete_item
+from todo_app.data.user import User
 from todo_app.flask_config import Config
 from todo_app.view import ViewModel
 
+def get_access_token_query_params():
+    return {
+        'client_id' : os.getenv('CLIENT_ID'),
+        'client_secret' : os.getenv('CLIENT_SECRET')
+    }
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object(Config())
+    login_disabled = os.getenv('LOGIN_DISABLED') == 'True'
+    app.config['LOGIN_DISABLED'] = login_disabled
 
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        return redirect(f"https://github.com/login/oauth/authorize?client_id={os.getenv('CLIENT_ID')}", code=302)
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        return User(user_id)
+
+    login_manager.init_app(app)
+
+    user_roles = {
+        'None' : 'reader', # for any user not actually logged in (i.e. anonymous users) are given reader roles unless login has been disabled
+        '109098469' : 'writer'
+    }
+
+    def get_current_user_role():
+        if (login_disabled):
+            return 'writer'
+        return user_roles.get(current_user.id)
```

Comment:
Minor, but personally I'm a fan of putting this sort of function as a method on the User class, but this will involve a few tweaks so not necessarily worth it here